[frontend] Add no data available placeholder text in empty simulations charts

isselparra opened the pull request:
### Proposed changes
- Add no data available placeholder text in empty scenario results chart

### Related issues

Close https://github.com/OpenBAS-Platform/openbas/issues/1371

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- ~~[ ] I wrote test cases for the relevant uses case~~
- ~~[ ] I added/update the relevant documentation (either on github or on notion)~~
- [x] Where necessary I refactored code to improve the overall quality


Before:
![image](https://github.com/user-attachments/assets/4a07673c-b8d4-4abe-882f-809e96c621fe)

After:
![image](https://github.com/user-attachments/assets/38b08074-faf9-440c-af15-d0e0f3a7d5a9)

Dimfacion commented:
Seems to be working but I'm unsure that's what we want. 
@jborozco can you confirm that you want an empty block with just "No data to display" ?

jborozco commented:
Hello viewed on Teams with Issel, We want to have the text but also a background to look nicer

isselparra commented:
PR updated @jborozco @Dimfacion

guillaumejparis commented:
Build failed https://drone.filigran.io/OpenBAS-Platform/openbas/7119/1/7 
I think you maybe just missed some intellij plugin for eslint

isselparra commented:
@guillaumejparis the plugin was already installed, an intellij config needed to be updated

Dimfacion commented on openbas-front/src/utils/Charts.js:
```diff
@@ -273,6 +275,8 @@ export const verticalBarsChartOptions = (
   isResult = false,
   isFakeData = false,
   max = undefined,
+  isChartAPlaceholder = false,
```
Can we maybe use the existing isFakeData property instead of adding a new one ?

isselparra replied:
If we use `isFakeData` directly we are going to also impact the `Simulations` chart in the home dashboard.
The scope described in the issue is only to impact the `Simulations results` chart at the scenario dashboard level.

Should we also modify the `Simulations` chart ? cc @jborozco

isselparra replied:
Discussed with @jborozco, we are updating also the `Simulations` chart

Dimfacion commented:
I sometimes have an issue where 'No data to display' appears in front of the graph even though there are datas. 
![image](https://github.com/user-attachments/assets/7f715f28-1300-458a-bd8b-99eae746f5b4)
Disabling the cache seems to fix the issue so maybe a problem with the store ?

isselparra commented:
@Dimfacion I corrected the problem with the "No data to display" text appearing event when the data is present.

I added a loader and I don't see the issue anymore.

Let me know if it's OK on your side :pray:

guillaumejparis reviewed (approved):
LGTM
